core/Tracker/BotRequest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license https://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+declare(strict_types=1);
+
+namespace Piwik\Tracker;
+
+use Piwik\Common;
+use Piwik\Container\StaticContainer;
+
+/**
+ * Class used to handle a Bot request.
+ */
+class BotRequest
+{
+    use RequestHandlerTrait;
+
+    /**
+     * @var  Request
+     */
+    protected $request;
+
+    /**
+     * @var BotRequestProcessor[]
+     */
+    protected $botRequestProcessors;
+
+    /**
+     * @var RequestProcessor[]
+     */
+    protected $requestProcessors;
+
+    public function __construct()
+    {
+        $requestProcessors          = StaticContainer::get('Piwik\Plugin\RequestProcessors');
+        $this->requestProcessors    = $requestProcessors->getRequestProcessors();
+        $this->botRequestProcessors = $requestProcessors->getBotRequestProcessors();
+    }
+
+    /**
+     * @param Request $request
+     */
+    public function setRequest(Request $request)
+    {
+        $this->request = $request;
+    }
+
+    public function handle()
+    {
+        $this->checkSiteExists($this->request);
+
+        /**
+         * For BC reasons we iterate over all visit request processors as well, to ensure a possible request manipulation is applied
+         * For Matomo 6 we should remove that and ensure plugins that also should manipulate bot requests implement a BotRequestProcessor for it
+         * @deprecated
+         */
+        foreach ($this->requestProcessors as $processor) {
+            Common::printDebug("Executing " . get_class($processor) . "::manipulateRequest()...");
```

Comment:
Replace these calls in the class with LoggerInterface instance as recently changed elsewhere?